zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceMigrationMigrateProcessor.java

```diff
@@ -274,10 +285,61 @@ private void tryMigrateElementInstance(
                         .setBpmnProcessId(targetProcessDefinition.getBpmnProcessId())
                         .setTenantId(elementInstance.getValue().getTenantId())));
 
+    final var context = new BpmnElementContextImpl();
+    context.init(elementInstance.getKey(), elementInstanceRecord, elementInstance.getState());
+    final var targetElement =
+        targetProcessDefinition
+            .getProcess()
+            .getElementById(targetElementId, ExecutableActivity.class);
+    final List<DirectBuffer> subscribedMessageNames = new ArrayList<>();
+    catchEventBehavior
+        .subscribeToEvents(
+            context,
+            targetElement,
+            catchEvent -> {
+              final var element = catchEvent.element();
+              final String targetCatchEventId = BufferUtil.bufferAsString(element.getId());
+              if (sourceElementIdToTargetElementId.containsValue(targetCatchEventId)) {
+                throw new ProcessInstanceMigrationPreconditionFailedException(
+                    """
+                    Expected to migrate process instance '%s' \
+                    but active element with id '%s' is mapped to element with id '%s' \
+                    that must be subscribed to mapped catch event with id '%s'. \
+                    Migrating active elements with mapped catch events is not possible yet."""
+                        .formatted(
+                            processInstanceKey, elementId, targetElementId, targetCatchEventId),
+                    RejectionType.INVALID_STATE);
+              }
+              if (element.isMessage()) {
+                requireNoSubscriptionForMessage(
+                    elementInstance,
+                    catchEvent.messageName(),
+                    elementInstanceRecord.getTenantId(),
+                    targetCatchEventId);
+                subscribedMessageNames.add(catchEvent.messageName());
+              }
+              return true;
+            })
+        .ifLeft(
+            failure -> {
+              throw new ProcessInstanceMigrationPreconditionFailedException(
+                  """
+                  Expected to migrate process instance '%s' \
```

Comment:
🔧 Falling into left means an unexpected error occurred, right? I think, the error message should be updated based on that.

Reply:
It's not really an unexpected error. It actually only happens on failing a related expression evaluation. The error message contains a clear description of what went wrong. You can see the error message in a09a1fa653825fdb4ae9d733b50a5634e0ff9ece.

❓ Is there anything you would change about it?

>Expected to migrate process instance '%s' but active element with id 'A' is mapped to element with id 'B' that must be subscribed to a message catch event. Failed to extract the correlation key for 'key': The value must be either a string or a number, but was 'NULL'. The evaluation reported the following warnings:
[NO_VARIABLE_FOUND] No variable found with name 'key'